R/delayed_choices.R

```diff
@@ -92,3 +105,57 @@ last_choice <- function() {
     class = c("delayed_choices", "delayed_data")
   )
 }
+
+#' @export
+#' @rdname delayed_choices
+first_choices <- function(n) {
+  checkmate::assert_count(n, positive = TRUE)
+  checkmate::assert_true(n > 1, .var.name = "n > 1")
+  structure(
+    function(x) {
+      if (inherits(x, "delayed_choices")) {
+        x
+      } else if (length(x) == 0L) {
+        x
+      } else if (is.atomic(x)) {
+        x[1:min(n, length(x))]
+      } else if (inherits(x, "delayed_data")) {
+        if (is.null(x$subset)) return(x)
+        original_fun <- x$subset
+        added_fun <- function(x) x[1:min(n, length(x))]
+        x$subset <- function(data) {
+          added_fun(original_fun(data))
+        }
+        x
+      }
+    },
+    class = c("multiple_choices", "delayed_choices", "delayed_data")
+  )
+}
+
+#' @export
+#' @rdname delayed_choices
+last_choices <- function(n) {
+  checkmate::assert_count(n, positive = TRUE)
+  checkmate::assert_true(n > 1, .var.name = "n > 1")
+  structure(
+    function(x) {
+      if (inherits(x, "delayed_choices")) {
+        x
+      } else if (length(x) == 0L) {
+        x
+      } else if (is.atomic(x)) {
+        x[max(1, (length(x) - (n - 1))):length(x)]
```

Comment:
Yyyy, what is it, C++? ;)

```suggestion
        utils::tail(x, n)
```

Reply:
I didn't want to add to `Imports`. I'll rewrite with `head` and `tail`.